loader/include/Geode/utils/Task.hpp

```diff
@@ -1016,7 +1097,10 @@ namespace geode {
                     static_cast<void*>(new EventListener<Task<T, P>>(
                         [handle](auto* event) {
                             if (event->getValue()) {
-                                handle.resume();
+                                // resume on main thread
+                                queueInMainThread([handle]() {

```

Comment:
this is not needed, event listener callback is already guaranteed to be on the main thread

Reply:
fair enough

Reply:
fixed